### PR TITLE
UrlHaus enchantment

### PR DESCRIPTION
**Pull request by AradCarmi**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/40676
## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by dantavori — CHANGES_REQUESTED** [review]
amazing work! see my notes.
in addition:
1. please use typing - e.g.  change `def domain_calculate_score(blacklist):` to `def domain_calculate_score(blacklist: List[str]) -> Tuple[int, str]:`.
2. In each reputation command you have a bit different convention/standard (e.g. in domain/file you don't use DBotScore object and the outputs_key_field, etc). Let's stick into one convention in all three commands (I suggest the convention in the URL command which is the most updated one).

File: Packs/URLHaus/Integrations/URLHaus/URLHaus.py [hunk 1]
```diff
@@ -86,137 +84,386 @@ def test_module(**kwargs):
...
+
+        Args:
+            urlhaus_data(dict).
+
+        Returns:
+            Tags(list).
+    """
+    tags = urlhaus_data.get('Tags', [])
+    if urlhaus_data.get('Threat'):
+        tags.append(urlhaus_data.get('Threat'))
```
[the remark this answers is not in the document]
Comment (AradCarmi) [reply Packs/URLHaus/Integrations/URLHaus/URLHaus.py:232]: I've already asked her and she told me to keep it under URL.Tags .

File: Packs/URLHaus/Integrations/URLHaus/URLHaus.py [hunk 2]
```diff
@@ -269,53 +519,72 @@ def domain_command(**kwargs):
...
+
+        Args:
+             urls(list)- url associated with file, sig(str)- file signature, file(str)- hash, params(dict).
+
+        Returns:
+            Relationships(list).
+    """
+    relationships = []
+    if kwargs.get('create_relationships'):
+        relationships.append(EntityRelationship(
```
[the remark this answers is not in the document]
Comment (AradCarmi) [reply Packs/URLHaus/Integrations/URLHaus/URLHaus.py:561]: Each file should have a signature so I guess the answer is yes.

File: Packs/URLHaus/Integrations/URLHaus/URLHaus.py [hunk 3]
```diff
@@ -86,137 +84,386 @@ def test_module(**kwargs):
...
+
+        Args:
+            bl_status(str)- blacklist status, tags(list).
+
+        Returns:
+            append to tags list(passed by ref).
+    """
+    if bl_status:
+        status_prefix = bl_status.split('_')[0] if bl_status.endswith('domain') else \
+            bl_status.split('_')[-1] if bl_status.startswith('abused') else ''
```
[the remark this answers is not in the document]
Comment (meital-aizen) [reply Packs/URLHaus/Integrations/URLHaus/URLHaus.py:466]: @dantavori After consulting, we decided that for tags that start with `abuse`, we can just save it as is (no need to remove any words).

File: Packs/URLHaus/Integrations/URLHaus/README.md [hunk 4]
```diff
@@ -61,83 +70,28 @@ Retrieves URL information from URLhaus.
...
         "Reliability": "C - Fairly reliable",
-        "Score": 1,
+        "Score": 0,
         "Type": "url",
         "Vendor": "URLhaus"
     },
     "URL": {
         "Data": "http://example.com/VMYB-ht_JAQo-gi/INV/99401FORPO/20673114777/US/Outstanding-Invoices/"
-    },
-    "URLhaus": {
```
Comment (dantavori) [inline Packs/URLHaus/Integrations/URLHaus/README.md:81]: now it's too little information :(
what I meant previously was that you should edit this manually and remove some of the output, so it won't be too long for the reader (but they still might want to see an informative example).

**Review by dantavori — CHANGES_REQUESTED** [review]
awesome 👏 
1. in addition to the new comments, see 2 comments that I unresolved from the previous CR.
2. please attach to the PR a screenshot of a passing run of the test playbook
3. we will need to add a support for providing multiple values in a reputation command (let's discuss this)
4. please schedule a demo for next week (invite Meital too)

File: Packs/URLHaus/Integrations/URLHaus/URLHaus.py [hunk 5]
```diff
@@ -86,236 +85,503 @@ def test_module(**kwargs):
...
-                'Contents': domain_information,
-                'HumanReadable': human_readable,
-                'HumanReadableFormat': formats['markdown'],
-                'EntryContext': ec
-            })
-        elif domain_information['query_status'] == 'invalid_host':
-            human_readable = f'## URLhaus reputation for {domain}\n' \
-                f'Invalid domain!'
+        Returns:
+            void.
```
[the remark this answers is not in the document]
Comment (AradCarmi) [reply Packs/URLHaus/Integrations/URLHaus/URLHaus.py:448]: Didn't understand the suggestion.

[the remark this answers is not in the document]
Comment (dantavori) [reply Packs/URLHaus/Integrations/URLHaus/URLHaus.py:448]: no need to provide "Returns:" when nothing is returned.

**Comment by AradCarmi** [comment]
> awesome 👏
> 
> 1. in addition to the new comments, see the comment that I unresolved from the previous CR.
> 2. please attach to the PR a screenshot of a passing run of the test playbook
> 3. we will need to add a support for providing multiple values in a reputation command (let's discuss this)
> 4. please schedule a demo for next week (invite Meital too)

File: Packs/URLHaus/Integrations/URLHaus/URLHaus.py [hunk 6]
```diff
@@ -86,236 +85,503 @@ def test_module(**kwargs):
...
+    """
+    if bl_status:
+        tag_to_add = bl_status.replace('_domain', '') if bl_status.endswith('domain') else \
+            bl_status if bl_status.startswith('abused') else ''
```
[the remark this answers is not in the document]
Comment (dantavori) [reply Packs/URLHaus/Integrations/URLHaus/URLHaus.py:462]: why did we resolve this?

File: Packs/URLHaus/Integrations/URLHaus/URLHaus.py [hunk 7]
```diff
@@ -86,236 +85,503 @@ def test_module(**kwargs):
...
+    if surbl:
+        if surbl == 'listed':
+            return Common.DBotScore.BAD, "The queried Domain is listed on SURBL"
+    if spamhaus_dbl:
+        if spamhaus_dbl == 'not_listed':
+            return Common.DBotScore.NONE, "The queried Domain is not listed on Spamhaus DBL"
+    if surbl:
+        if surbl == 'not_listed':
+            return Common.DBotScore.NONE, "The queried Domain is not listed on SURBL"
+    return Common.DBotScore.GOOD, "There is no information about Domain in the blacklist"
```
[the remark this answers is not in the document]
Comment (dantavori) [reply Packs/URLHaus/Integrations/URLHaus/URLHaus.py:136]: why did you resolve?

File: Packs/URLHaus/Integrations/URLHaus/URLHaus.py [hunk 8]
```diff
@@ -86,330 +85,624 @@ def test_module(**kwargs):
...
+
+def url_command(params: dict):
+    """
+        Split urls and call run_url_command on each of them.
+
+         Args:
+            params (dict): The integration params.
+    """
+    urls = demisto.args().get('url', '')
+    for url in urls.split(','):
```
Comment (dantavori) [inline Packs/URLHaus/Integrations/URLHaus/URLHaus.py:420]:
```suggestion
    for url in argToList(urls):
```
url arg might already be a list given to the argument from the context:
`!url url=${Path.to.URLs}`
argToList() handles this. please change in other commands too.

**Review by dantavori — CHANGES_REQUESTED** [review]
almost there!
see my last minor notes. in addition: please change in the yml in every reputation command the `isArray` value from `false` to `true` (e.g. in line 72 for url command).
explanation is here: https://xsoar.pan.dev/docs/integrations/code-conventions#ioc-reputation-commands

File: Packs/URLHaus/Integrations/URLHaus/URLHaus.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (AradCarmi) [reply Packs/URLHaus/Integrations/URLHaus/URLHaus.py:136]: The design document defines that in any other case the DBot Score should be benign.

**Comment by ShirleyDenkberg** [comment]
@meital-aizen @dantavori Doc review completed.

**Pull request merged**